draft-zip.md

```diff
@@ -0,0 +1,97 @@
+
+# Motivation
+
+While ZIP-XXX (Establishing the Zcash Sustainability Fund) describes a method by which funds can be added to the Zcash Sustainability Fund by a voluntary `ZSF_DEPOSIT` transaction field. The default value of this field is zero and it is left up to the app and wallet implementers to make use of it.
+
+This ZIP takes a much more explicit and non-optional approach, mandating at the protocol level that 50% of transaction fees be deposited into the ZSF As noted above, implementing this change allows the ZSF to accrue value earlier and contribute to future network sustainability.
+
+This system currently looks something like this:
+
+At Every New Block:
+- `ZSF_DEPOSIT` amount is deposited into the `ZSF_BALANCE_AFTER[h]`
+- Miner rewards come from `ZSF_BALANCE_AFTER[h]`
+- Transaction fees (inputs - outputs) paid to miner
+
+After the features described in this ZIP are activated (changed parts in bold):
+
+At Every New Block:
+- `ZSF_DEPOSIT` amount is deposited into the `ZSF_BALANCE_AFTER[h]`
+- Miner rewarded from `ZSF_BALANCE_AFTER[h]`
+- 50% of transaction fees (inputs - outputs) paid to miner
+- 50% of transaction fees deposited into `ZSF_BALANCE_AFTER[h]`
+
+This has a multitude of benefits:
+
+1. **Network Sustainability**: This mechanism involves temporarily reducing the supply of ZEC similar to asset burning in Ethereum’s EIP-1559, but with potential long-term sustainability benefits as the redistribution of deposits contributes to issuance rewards and network development, making it an attractive option for current and future Zcash users.
+2. **Ecosystem Benefits of Longer Time Horizons**: A reliable and long-term functioning Zcash blockchain allows users to make secure long-term plans, leading to a sustainable and virtuous adoption cycle, rather than being influenced by short-term trends.
+3. **Incentivizing Transaction Inclusion**: By providing a 10% share of transaction fees to miners, this ZIP maintains incentives for miners to prioritize including transactions in their blocks. This helps ensure the efficient processing of transactions and supports a robust and responsive network.
+4. **Future-Proofing the Network**: Diverting transaction fees into the `ZSF_BALANCE_AFTER[h]` is a forward-looking approach that prepares the Zcash network for future challenges and opportunities. It establishes a financial buffer that can be instrumental in addressing unforeseen issues and seizing strategic advantages as the Zcash ecosystem evolves.
+
+# Specification
+
+This ZIP only proposes a single modification to the transaction fees:
+1. Keep the current destination of 10% of the fees untouched, but route 50% of the fees back to `ZSF_BALANCE_AFTER[h]`
+
+Please note that a network upgrade is required for this work to be fully implemented.
+
+## Transaction fee routing requirements
+
+- For each transaction, 50% of the total fee MUST be paid to the `ZSF_BALANCE_AFTER[h]`
+- The minimum of the 50% fee MUST equal 1 zatoshi or more.
```

Comment:
1. How is the first requirement specified precisely? I suggest adding a consensus rule that requires the `zsfDeposit` field in the coinbase transaction to be at least 50% of the transaction fee. This needs to be defined using a mathematical formula.
2. The second requirement is impossible because zero-fee transactions are valid. It's also unclear whether the 50% is the miner's or ZSF 50%. Instead, precisely define rounding by using the `ceiling()` or `floor()` functions in the mathematical formula.

For example, here are the current consensus rules for funding streams:
> The coinbase transaction at block height height MUST contain at least one output per funding stream fs active at height, that pays fs.Value(height) zatoshi in the prescribed way to the stream’s recipient address represented by fs.AddressListfs.AddressIndex(height).

https://zips.z.cash/protocol/protocol.pdf#fundingstreams

And miner fees (the miner can spent whatever is left after the funding streams however they like):
> The total value in zatoshi of transparent outputs from a coinbase transaction, minus vbalanceSapling, minus vbalanceOrchard, MUST NOT be greater than the value in zatoshi of block subsidy plus the transaction fees paid by transactions in this block.

https://zips.z.cash/protocol/protocol.pdf#txnconsensus

Here's one possible way to make the rule precise (please choose the rounding you want here):

```suggestion
## Transaction fee routing requirements

- For each transaction, 50% of the total fee MUST be paid to the ZSF
- Any fractions are rounded in favour of the miner _TODO ZIP owners: decide if you want rounding to favour the ZSF here_

### Consensus Rule Changes

The coinbase transaction at block height `height` MUST have a  `zsfDeposit(height)` that is greater than or equal to `floor(TransactionFees(height)) / 2)`, where `TransactionFees(height)` is the sum of the the remaining value in the transparent transaction value pool of the non-coinbase transactions in the block at `height`.

_TODO ZIP owners: if you want rounding to favour the ZSF, use ceiling here_

TODO ZIP Editors:
- work out how to deal with pre-v6 transactions which don't have the zsfDeposit() field. For example, by requiring the remaining value in the transparent transaction value pool of a coinbase transaction to be greater than or equal to 50% of the fee
- consider imposing this requirement on v6 transactions instead of an explicit ZSF_DEPOSIT field requirement
```

Reply:
> 2\. It's also unclear whether the 50% is the miner's or ZSF 50%.

This kind of confusion could happen in specification, coinbase transaction creation, validation, or testing. A simple fix for this issue is using a split where the fractions aren't equal. 51/49 would be sufficient, but maybe the security model will suggest another split.

See my final paragraph on:
https://forum.zcashcommunity.com/t/the-zcash-posterity-fund/42703/104